ci-operator/step-registry/cucushift/installer/rehearse/gcp/cases/valid-confidential-computing/provision/cucushift-installer-rehearse-gcp-cases-valid-confidential-computing-provision-commands.sh

```diff
@@ -0,0 +1,302 @@
+  fi
+
+  set +o errexit
+  if echo "${unsupported_cc_types}" | grep -qw "${cc_type1}"; then
+    if [[ "${expected_err_msg}" != "" ]]; then
+      expected_err_msg="${expected_err_msg}, "
+    fi
+    expected_err_msg="${expected_err_msg}controlPlane.platform.gcp.type: Invalid value: \"${instance_type}\": Machine type do not support ${cc_type1}. Machine types supporting ${cc_type1}: ${cc_type_machine_series_dict[${cc_type1}]}"
+  fi
+
+  if [[ "${on_host_maintenance}" != "Terminate" ]]; then
+    expected_err_msg="${expected_err_msg}, compute[0].platform.gcp.onHostMaintenance: Invalid value: \"${on_host_maintenance}\": onHostMaintenace must be set to Terminate when confidentialCompute is ${cc_type2}"
+  fi
+
+  if echo "${unsupported_cc_types}" | grep -qw "${cc_type2}"; then
+    if [[ "${expected_err_msg}" != "" ]]; then
+      expected_err_msg="${expected_err_msg}, "
+    fi
+    expected_err_msg="${expected_err_msg}compute[0].platform.gcp.type: Invalid value: \"${instance_type}\": Machine type do not support ${cc_type2}. Machine types supporting ${cc_type2}: ${cc_type_machine_series_dict[${cc_type2}]}"
+  fi
+  set -o errexit
+}
+
+function random_choice()
+{
+  local -r cc_types_str="$1"
+  local array_len selected_index
+
+  IFS=' ' read -r -a cc_types_array <<< "${cc_types_str}"
+  array_len="${#cc_types_array[@]}"
+  selected_index=$(( $RANDOM%($array_len) ))
+  echo "${cc_types_array[${selected_index}]}"
+}
+
+function create_manifests()
+{
+  local -r instance_type="$1"
+  local -r cc_types_str="$2"
+  local -r on_host_maintenance="$3"
+  local -r testing_scenario_num="$4"
+  local -r success_flag="$5"
+  local cluster_name install_dir compute_cc_type control_plane_cc_type default_machine_platform_cc_type result tmp_output use_default_flag
+
+  tmp_output=$(mktemp)
+  cluster_name="${CLUSTER_PREFIX}${testing_scenario_num}"
+  install_dir="/tmp/${cluster_name}"
+  mkdir -p "${install_dir}" 2>/dev/null
+  create_install_config "${cluster_name}" "${install_dir}"
+
+  compute_cc_type=$(random_choice "${cc_types_str}")
+  control_plane_cc_type=$(random_choice "${cc_types_str}")
+  default_machine_platform_cc_type=$(random_choice "${cc_types_str}")
+
+  use_default_flag=false
+  if [[ $RANDOM%100 -gt 30 ]]; then
+    echo "$(date -u --rfc-3339=seconds) - Use 'compute_cc_type' & 'control_plane_cc_type'..."
+    cat > "${patch}" << EOF
+compute:
+- name: worker
+  platform:
+    gcp:
+      type: ${instance_type}
+      confidentialCompute: ${compute_cc_type}
+      onHostMaintenance: ${on_host_maintenance}
+controlPlane:
+  name: master
+  platform:
+    gcp:
+      type: ${instance_type}
+      confidentialCompute: ${control_plane_cc_type}
+      onHostMaintenance: ${on_host_maintenance}
+EOF
+  else
+    echo "$(date -u --rfc-3339=seconds) - Use 'default_machine_platform_cc_type'..."
+    use_default_flag=true
```

Comment:
Shall we take the case that combines both `platform.gcp.defaultMachinePlatform`, `compute[0].platform.gcp` and `controlPlane.platform.gcp` into consideration ?